applications/FluidDynamicsApplication/python_scripts/distance_reinitialization.py

```diff
@@ -0,0 +1,136 @@
+            "max_distance": 1.0,
+            "calculate_exact_distances_to_plane" : true
+        }""")
+        self.params.ValidateAndAssignDefaults(default_settings)
+
+    def _GetVariationalCalculateExactDistancesFlag(self):
+        if self.params["calculate_exact_distances_to_plane"].GetBool():
+            return KratosMultiphysics.VariationalDistanceCalculationProcess2D.CALCULATE_EXACT_DISTANCES_TO_PLANE
+        else:
+            return KratosMultiphysics.VariationalDistanceCalculationProcess2D.CALCULATE_EXACT_DISTANCES_TO_PLANE.AsFalse()
+
+    def _GetParallelCalculateExactDistancesFlag(self):
+        if self.params["calculate_exact_distances_to_plane"].GetBool():
+            return KratosMultiphysics.ParallelDistanceCalculator2D.CALCULATE_EXACT_DISTANCES_TO_PLANE
+        else:
+            return KratosMultiphysics.ParallelDistanceCalculator3D.CALCULATE_EXACT_DISTANCES_TO_PLANE.AsFalse()
+
+    def _GetLinearSolver(self):
+        linear_solver_configuration = self.params["linear_solver_settings"]
+        return linear_solver_factory.ConstructSolver(linear_solver_configuration)
+
+    def _ConstructVariationalProcess(self, maximum_iterations, linear_solver, process_flag):
+        if self.model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE] == 2:
+            return KratosMultiphysics.VariationalDistanceCalculationProcess2D(
+                self.model_part,
+                linear_solver,
+                maximum_iterations,
+                process_flag)
+        else:
+            return KratosMultiphysics.VariationalDistanceCalculationProcess3D(
+                self.model_part,
+                linear_solver,
+                maximum_iterations,
+                process_flag)
+            
+    def _ConstructParallelProcess(self):
+        if self.model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE] == 2:
+            return KratosMultiphysics.ParallelDistanceCalculator2D()
+        else:
+            return KratosMultiphysics.ParallelDistanceCalculator3D()
+
+    def ReinitializeDistances(self):
+        if (self._reinitialization_type == "variational"):
+            self.distance_reinitialization_process.Execute()
+
+        elif (self._reinitialization_type == "parallel"):
+            layers = self.params["max_layers"].GetInt()
+            max_distance = self.params["max_distance"].GetDouble()
+            process_flag = self._GetParallelCalculateExactDistancesFlag()
+            self.distance_reinitialization_process.CalculateDistances(
+                self.model_part,
+                KratosMultiphysics.DISTANCE,
+                KratosMultiphysics.NODAL_AREA,
+                layers,
+                max_distance,
+                process_flag)
+
+        if (self._reinitialization_type != "none"):
+            KratosMultiphysics.Logger.PrintInfo("DistanceReinitialization", "Redistancing process is finished.")
+
+
+class ParallelDistanceReinitialization(DistanceReinitialization):
```

Comment:
I recommend moving this to a separate file to avoid the conditional imports

Reply:
I was seeing that one coming : D

Reply:
Having to fight with trilinos at least one a month (still after all the years!!!) I will keep pointing this out xD